src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -152,51 +154,117 @@ object DepositHandler {
     }
   }
 
-  def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
-    log.debug(s"[$id] Checking if bag contains fetch.txt")
-    getFetchTxt(bagitDir).foreach(fetchText =>
-      { checkFetchTxtUrls(fetchText)
-        resolveFetchTxtFiles(bagitDir, fetchText)
-      }
-    )
-  }
-
   private def getFetchTxt(bagitDir: File)(implicit id: String): Option[FetchTxt] = Option {
     getBagFromDir(bagitDir).getFetchTxt
   }
 
-  private def checkFetchTxtUrls(fetchText: FetchTxt)(implicit id: String): Unit = {
+  def checkFetchTxtUrls(bagitDir: File)(implicit id: String): Try[Unit] = Try {
     log.debug(s"[$id] Checking validity of urls in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      checkUrlValidity(item.getUrl))
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      fetchText.asScala.foreach(item =>
+      checkUrlValidity(item.getUrl)))
   }
 
   private def checkUrlValidity(url: String)(implicit id: String): Unit = {
     // check if the url is syntactically correct
     try {
       new URL(url)
     } catch {
-      case e: MalformedURLException => throw new InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
+      case e: MalformedURLException => throw InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
     }
     // check if the url complies with the allowed url-structure
     val urlPattern = Pattern.compile(SwordProps("url-pattern"))
     if (!urlPattern.matcher(url).matches)
-      throw new InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
+      throw InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
   }
 
-  private def resolveFetchTxtFiles(bagitDir: File, fetchText: FetchTxt)(implicit id: String): Unit = {
-    log.debug(s"[$id] Resolving files in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      Using.urlInputStream(new URL(item.getUrl)).foreach(src =>
-        Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename))))
-  }
-
-  def checkBagValidity(bagitDir: File)(implicit id: String): Try[Unit] = {
+  def checkBagValidity(bagitDir: File)(implicit id: String) = {
     log.debug(s"[$id] Verifying bag validity")
+
+    val fetchTexts = getFetchTxt(bagitDir).map(fetchText => fetchText.asScala).getOrElse(List()).toList
+    val fetchTextsBagStore = fetchTexts.filter(_.getUrl.startsWith(baseUrl))
+    val fetchTextsOther = fetchTexts diff fetchTextsBagStore
```

Comment:
Consider removing this temp. It is not really bad, but we use it only once and the `diff` expression really says it all.